In [39]:
# Importing Libary
import pandas as pd
import numpy as np
from scipy import stats

In [41]:
# Load the dataset
df = pd.read_csv(r"C:\Users\HP\Downloads\property.csv")

In [43]:
# Convert date column to datetime format
df['Date'] = pd.to_datetime(df['Date'], format='%d/%m/%Y')

In [45]:
#1 Hypothesis Testing for Altona
altona_prices = df[df['Suburb'] == 'Altona']['Price'].dropna()
mean_price = 800000  # Hypothesized mean price

# Perform one-sample t-test
t_stat, p_value = stats.ttest_1samp(altona_prices, mean_price)
print(f"T-Statistic: {t_stat}, P-Value: {p_value}")

T-Statistic: 1.0277020770199676, P-Value: 0.307483271305555


#Hypothesis Testing for Altona Property Prices
#Since the p-value (0.307) is greater than 0.05, we fail to reject the 
#hypothesis that the typical property price is $800,000.

In [69]:
# 2 Compare means for Summer vs Winter in 2016
df_2016 = df[df['Date'].dt.year == 2016]

winter_months = [10, 11, 12, 1, 2, 3]
summer_months = [4, 5, 6, 7, 8, 9]

winter_prices = df_2016[df_2016['Date'].dt.month.isin(winter_months)]['Price'].dropna()
summer_prices = df_2016[df_2016['Date'].dt.month.isin(summer_months)]['Price'].dropna()

# Perform independent t-test
t_stat_seasonal, p_value_seasonal = stats.ttest_ind(winter_prices, summer_prices, equal_var=False)
print(f"Seasonal T-Test: {t_stat_seasonal:3f}, P-Value: {p_value_seasonal:3f}")

Seasonal T-Test: 3.921101, P-Value: 0.000090


#Comparison of Seasonal Property Prices in 2016
#Since the p-value is less than 0.05, there is a significant difference in property prices between summer and winter months.

In [49]:
# 3 Probability calculations for Abbotsford
df_abbotsford = df[df['Suburb'] == 'Abbotsford']

# Probability of 3 out of 10 properties having no car parking
p_no_car = (df_abbotsford['Car'].fillna(0) == 0).mean()
prob_3_no_car = stats.binom.pmf(3, 10, p_no_car)
print(f"Probability of 3 out of 10 properties without parking: {prob_3_no_car:.3f}")

Probability of 3 out of 10 properties without parking: 0.260


In [51]:
# 4 Probability of finding a property with 3 rooms
p_3_rooms = (df_abbotsford['Rooms'] == 3).mean()
print(f"Probability of 3-room property: {p_3_rooms:.3f}")

Probability of 3-room property: 0.357


In [53]:
# 5 Probability of finding a property with 2 bathrooms
p_2_bathrooms = (df_abbotsford['Bathroom'] == 2).mean()
print(f"Probability of 2-bathroom property: {p_2_bathrooms:.3f}")

Probability of 2-bathroom property: 0.339
